### ISSUE TITLE
Bug: Plugin always needs to save twice to store file path 

### ISSUE THREAD

**Issue by sjaehn** [falkTX/Carla#1345]
Hi,

I found a bug in my LV2 plugins that seems to be caused by Carla. Within the plugin state:save and state:restore. It seems like abstractPath / absolutePath doesn't  work on the first call. But all works fine from the second one.

First try to save (Carla's save button) fails:
```
Project directory not set, cannot map abstractPath '.'
Project directory not set, cannot map absolutePath /home/sven/Downloads/oops44100.wav
BJumblr.lv2: Can't generate abstr_path from /home/sven/Downloads/oops44100.wav
Carla assertion failure: "size > 0" in file ../utils/CarlaMathUtils.hpp, line 109
```
(the BJumblr message is generated by the plugin)

Second one works:
```
Creating symlink for '/home/sven/Downloads/oops44100.wav' in '/home/sven/Carla/B.Jumblr'
BJumblr.lv2: Save abstr_path:oops44100.wav
Carla assertion failure: "size > 0" in file ../utils/CarlaMathUtils.hpp, line 109
```

It's highly reproducible on my system (Ubuntu 20.04, Carla 2.3.0-alpha2).

**Comment by falkTX** [comment]
@sjaehn can you try this with latest git/main of carla?
Due to https://github.com/sjaehn/BJumblr/issues/28 I am not able to test this myself.

**Comment by falkTX** [comment]
I was able to reproduce this with a different plugin.
Fixed in 5f3741c5d81672c4c65c7f02e499078e90e7a64c
